### PR TITLE
bsp: tf-a-fio: add correct fix for assert issue in TF-A

### PR DESCRIPTION
**Pull request by igoropaniuk**
This fixes false positive assers in TF-A BL2:
INFO:    BL2: Doing platform setup
INFO:    RAM: DDR3-DDR3L 32bits 533000kHz
INFO:    Memory size = 0x40000000 (1024 MB)
INFO:    BL2: Loading image id 31
ASSERT: drivers/io/io_block.c:272

Exception mode=0x00000016 at: 0x2fffa19c

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

### CODE REVIEW

File: meta-lmp-bsp/recipes-bsp/trusted-firmware-a/tf-a-fio/0007-feat-mmc-get-boot-partition-size.patch [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Date: Thu, 15 Sep 2022 09:51:21 +0300
+Subject: [PATCH 2/2] feat(mmc): get boot partition size
+
+The boot partition size of an eMMC is given in ext_csd register, at
+offset 226 (BOOT_SIZE_MULT), which has to be multiplied by 128kB.
+Add a helper function mmc_boot_part_size() to get this eMMC boot
+partition size.
+
+Signed-off-by: Yann Gautier <yann.gautier@st.com>
+Change-Id: I0e8e0fc9632f147fa1b1b3374accb78439025403
```
Comment (quaresmajose) [inline meta-lmp-bsp/recipes-bsp/trusted-firmware-a/tf-a-fio/0007-feat-mmc-get-boot-partition-size.patch:12]: Its good to have the Upstream-Status with the url link of the patch.
`Upstream-Status: Backport [url]`

File: meta-lmp-bsp/recipes-bsp/trusted-firmware-a/tf-a-fio/0007-feat-mmc-get-boot-partition-size.patch [hunk 2]
```diff
@@ -0,0 +1,57 @@
+From 6302a29a75447522c31d43616228055b8f3f2883 Mon Sep 17 00:00:00 2001
+From: Yann Gautier <yann.gautier@st.com>
+Date: Thu, 15 Sep 2022 09:51:21 +0300
+Subject: [PATCH 2/2] feat(mmc): get boot partition size
```
Comment (MrCry0) [inline meta-lmp-bsp/recipes-bsp/trusted-firmware-a/tf-a-fio/0007-feat-mmc-get-boot-partition-size.patch:4]: Just a comment: interesting that the patchset has the wrong order, the 1st commit uses a function introduced in the 2nd commit :)

**Review by MrCry0 — APPROVED** [review]

File: meta-lmp-bsp/recipes-bsp/trusted-firmware-a/tf-a-fio/0007-feat-mmc-get-boot-partition-size.patch (hunk 2 above)
Comment (igoropaniuk) [reply meta-lmp-bsp/recipes-bsp/trusted-firmware-a/tf-a-fio/0007-feat-mmc-get-boot-partition-size.patch:4]: oops, that was me who messed up :)
in the original MRs it's in correct order

**Review by MrCry0 — APPROVED** [review]

**Review by angolini — APPROVED** [review]

**Review by ricardosalveti — COMMENTED** [review]
LGTM

**Pull request merged**